services/py-api/src/database/db_manager.py

```diff
@@ -66,6 +67,35 @@ def get_collection(self, collection_name: str) -> AsyncIOMotorCollection:
         # https://pymongo.readthedocs.io/en/stable/tutorial.html#getting-a-collection
         return self._client[self._DB_NAME][collection_name]
 
+    @staticmethod
+    async def retry_db_operation[T](db_operation: Callable[..., Awaitable[T]], *args: Any, **kwargs: Any) -> T:
+        """
+        If there was a TransientTransactionError during execution, retries the db operation with exponential backoff up
+        to max_retries times. Having an exponential backoff increases our chances of success if there was a temporary
+        network issue.
+
+        Raises:
+            PyMongoError
+        """
+        max_retries = 3
+        delay = 1  # initial delay in seconds
+
+        # range is exclusive that's why we do max_retries + 1
+        for retry in range(1, max_retries + 1):
+            try:
+                return await db_operation(*args, **kwargs)
+            except PyMongoError as exc:
+                if exc.has_error_label("TransientTransactionError"):
+                    LOG.debug("Retrying transaction retry {}".format(retry))
+                    await sleep(delay)
+                    delay *= 2  # exponential backoff
+                    continue
+
+                # If the exception it's a non-retryable error re-raise it in order to be caught on an upper level
```

Comment:
Update the comment to include that it's either a non-retryable error or a transactional error which didn't manage to propagate

Reply:
Fixed in 062077d0b7deca169495ff24160e440924375f68. Keep in mind that the logic in `db_manager.py` might change as I am currently working on it, and the requsted change is made in the `transaction_manager.py`